dcos/nodeutil/util.go

```diff
@@ -364,6 +377,36 @@ func (d *dcosInfo) state(ctx context.Context) (state State, err error) {
 	return state, err
 }
 
+// TaskCanonicalID return a CanonicalID for a given task.
+func (d *dcosInfo) TaskCanonicalID(ctx context.Context, task string) (*CanonicalID, error) {
+	state, err := d.state(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, framework := range state.Frameworks {
+		for _, t := range framework.Tasks {
+			if t.Name != task && !strings.Contains(t.ID, task) {
+				continue
+			}
+
+			containerIDs, err := t.ContainerID()
+			if err != nil {
+				return nil, err
+			}
+
+			return &CanonicalID{
+				AgentID:      t.SlaveID,
+				FrameworkID:  t.FrameworkID,
+				ExecutorID:   t.ID,
```

Comment:
Is this intentionally `t.ID` and not `t.executorID`? If so, it could be misleading, since executor ID is a property of a task :/

Reply:
good point, here is the thing. With the docker/mesos runtime the executorID will be always empty, and we have to use ID

Reply:
I see. Would having a blank executor ID be harmful? Could we add the task ID as well?

Reply:
fixed